From fa0ba1e3a3f9d4a3970848648ced9c353a2295bb Mon Sep 17 00:00:00 2001
From: Istio Automation <istio-testing-bot@google.com>
Date: Tue, 28 Dec 2021 09:42:55 -0800
Subject: [release-1.12] Fix  label selector not select default pod (#36426)

* Fix label selector not select default pod

* Add release note

Co-authored-by: Kebe <kebe.liu@daocloud.io>
Co-authored-by: Kebe <mail@kebe7jun.com>
---
 istioctl/cmd/proxyconfig.go   |  3 +++
 releasenotes/notes/36182.yaml | 11 +++++++++++
 2 files changed, 14 insertions(+)
 create mode 100644 releasenotes/notes/36182.yaml

diff --git a/istioctl/cmd/proxyconfig.go b/istioctl/cmd/proxyconfig.go
index ba9dd72863..f8d45c9280 100644
--- a/istioctl/cmd/proxyconfig.go
+++ b/istioctl/cmd/proxyconfig.go
@@ -591,6 +591,9 @@ func logCmd() *cobra.Command {
 				if err != nil {
 					return err
 				}
+				if len(podNames) > 0 {
+					podName = podNames[0]
+				}
 			} else {
 				if podName, podNamespace, err = getPodName(args[0]); err != nil {
 					return err
diff --git a/releasenotes/notes/36182.yaml b/releasenotes/notes/36182.yaml
new file mode 100644
index 0000000000..b989655729
--- /dev/null
+++ b/releasenotes/notes/36182.yaml
@@ -0,0 +1,11 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+- 36182
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+- |
+  **Fixed** `istioctl pc log` command label selector not select default pod.
-- 
2.35.3

